External/FEXCore/Source/Interface/Core/OpcodeDispatcher/Flags.cpp

```diff
@@ -1003,17 +1004,21 @@ void OpDispatchBuilder::CalculcateFlags_RotateLeftImmediate(uint8_t SrcSize, Ord
 
   auto OpSize = SrcSize * 8;
 
+  auto NewCF = _Bfe(1, 0, Res);
   // CF
   {
     // Extract the last bit shifted in to CF
-    SetRFLAG<FEXCore::X86State::RFLAG_CF_LOC>(_Bfe(1, 0, Res));
+    SetRFLAG<FEXCore::X86State::RFLAG_CF_LOC>(NewCF);
   }
 
   // OF
   {
     if (Shift == 1) {
-      // OF is the top two MSBs XOR'd together
-      SetRFLAG<FEXCore::X86State::RFLAG_OF_LOC>(_Xor(_Bfe(1, OpSize - 1, Src1), _Bfe(1, OpSize - 2, Src1)));
+      // OF is the LSB and MSB XOR'd together
+      // OF is set to the XOR of the new CF bit and the most significant bit of the result
```

Comment:
Might be worth adding a note this is only well-defined for 1-bit shifts, since otherwise it's unclear why those specific bits are used.

Reply:
Updated the comment to say that OF is only architecturally defined for a 1-bit shift.